### DISCUSSION TITLE
Can I have a Rounded Sprite?

### DISCUSSION THREAD

**Discussion by humbleCoder1990** [pixijs/pixijs#7649] [Q&A]
I would like to use Sprite with tint.
Because It's much more performant than using Graphics.
But Can I have a rounded sprite like Rounded Rectangle?
Only to have rounded something, I don't want to go to Graphics.

**Comment by ivanpopelyshev** [comment]
Nope, no such element with pixi. best candidates:

1. sprite mask with antialiased circle ... bad perf, bad work with AA, you need MSAA or generateCanvasTexture
2. graphics beginTextureFill() , yeah, it will have AA if its on main framebuffer, but antialias on itself is slow
3. pixi-heaven sprite mask - much faster but still, you need a sprite of circle, antialiased!
4. smooth-graphics beginTextureFill with enabled AA - i dont remember whether i implemented it already, most probably it doesnt work
...
10. special sprite that you will sponsor via bounty mechanism

Also its needed by everyone who wants to clone `agar.io`, and yet no one actually cares to share their best solutions in open-source :)

**Reply by humbleCoder1990** [reply]
How about using canvas  as texture then?

![image](https://user-images.githubusercontent.com/87828192/127419225-278659ba-176e-4395-8e35-3f2e4f373107.png)

**Reply by ivanpopelyshev** [reply]
No problem :)

**Comment by miltoncandelero** [comment]
I might be misunderstanding the issue here but... Nine slice patch?

**Comment by eXponenta** [comment]
Main issues why graphics with rounded corners is slow - AA (very important for rounds) and a lot of vertices.
10px round emit ~6 vertices each corner instead of 1 for regular sprite.
